lib/vehicle.rb

```diff
@@ -1,18 +1,22 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_accessor :vin,
```

Comment:
We typically want to avoid attr_accessors as much as possible to maintain data integrity. Allowing other classes to mutate an object's attributes can lead to issues. For example, with this setup, there's nothing stopping our Facility class from setting the `year` attribute to 1234567890, which wouldn't make sense! For the purposes of this project, this is fine, but in applications on the job, we want to try to stick to read-only as much as possible. 

In this case with your current code, you really only need attr_accessors for :registration_date and :plate_type because those are the attributes you will be mutating in the facility class. 

if you wanted to avoid attr_accessors *completely*, the option would be to create some "setter" methods in this class called `assign_plate_type` and `assign_registration_date` and the Facility class could call those methods to make  changes to those attributes. The benefit to doing this is adding some controls in those methods (i.e. an if statement that makes sure that the plate_type is ONLY going to be set to the value of :ev, :antique, etc. Or a conditional to make sure the year is only set to an integer with 4 digits). We wouldn't expect you to add those controls into such a method as part of this project's requirements but I'm sharing this info so you know why we encourage that type of assignment rather than attr_accessors! Let me know if you have any questions about this